qt/widgets/plotting/src/PreviewPlot.cpp

```diff
@@ -414,6 +414,9 @@ bool PreviewPlot::eventFilter(QObject *watched, QEvent *evt) {
   case QEvent::Resize:
     stopEvent = handleWindowResizeEvent();
     break;
+  case QEvent::UpdateLater:
+    m_redrawOnPaint = true;
+    break;
```

Comment:
Just curious as to what this is doing

Reply:
There was a small issue introduced by disabling the interface. The PeakPicker tool on the matplotlib plot would sometimes disappear and not get redrawn after the interface had been disabled and then re-enabled.

This was caused by a race condition in which the order of the events captured by the event loop changed. If the `UpdateLater` event was processed after the `Paint` event, then the PeakPicker would not get redrawn at the appropriate moment. The solution was to ensure the peakpicker gets redrawn after the `UpdateLater` event is recieved.